subsys/dfu/dfu_target/src/dfu_target_mcuboot.c

```diff
@@ -150,6 +150,14 @@ int dfu_target_mcuboot_offset_get(size_t *out)
 	return err;
 }
 
+int dfu_target_mcuboot_erase(int slot, bool force)
+{
+	if (force) {
+		return boot_erase_img_bank(slot);
+	}
+	return 0;
+}
```

Comment:
Maybe add some log warning that if `force == false` we do nothing? Or some other return value? Because now it looks like erase went fine even if we didn't attempt an erase.

Reply:
Usually you shouldn't care if the slot is erased or not since the default behaviour is erase progressivly so for MCUBoot style upgrades it shouldn't matter too much if the slot is erased or not.
Can add a debug log but I think log warn is a bit too excessive.

Reply:
Hmm, yeah sure, I get your point. My thinking was more in the lines of how it is for modem_delta, where you check if the slot is dirty, and then erase (unless you force, then you always erase), which seems more in line with having a `force` option, that at least currently has the description: `Indicate whether the erase should be executed even if the area already reports as empty.`